MidCamp/midcamp
# Composer 2

# Description

- Updates Composer to version 2
  - Set in Dockerfile.cli via `selfupdate` command.  Per https://github.com/uselagoon/lagoon-images/issues/265 this is the recommended path at this time.
  - Targeted in `.lando.yml` file via `/usr/local/bin/composer` to prevent locally installed versions (which may still be on Composer 1) from being inadvertently used in local development
- Per https://www.drupal.org/docs/develop/using-composer/preparing-your-site-for-composer-2, updates the following packages in preparation for Composer 2:
  - composer/installers
  - cweagans/composer-patches
  - oomphinc/composer-installers-extender
  - drupal/console (specifically for drupal/console-extend-plugin)
- Updates `drupal/switch_page_theme` to the latest commit in `dev` for D9 support (removing patch that no longer applies)
- Updates `drupal/libraries` to beta release for D9 support (removing patch that no longer applies)

# Test instructions

- Rebuild your local environment via `lando rebuild -y`
- Run `lando composer -v` and confirm Composer 2 is in use
- Explore your local site and confirm it is still fully functional
